packages/opentelemetry-metrics/test/MeterProvider.test.ts

```diff
@@ -15,10 +15,25 @@
  */
 
 import * as assert from 'assert';
+import * as sinon from 'sinon';
 import { MeterProvider, Meter, CounterMetric } from '../src';
-import { NoopLogger } from '@opentelemetry/core';
+import {
+  NoopLogger,
+  notifyOnGlobalShutdown,
+  _invokeGlobalShutdown,
+} from '@opentelemetry/core';
+
+function _cleanupGlobalShutdownListeners() {
+  if (typeof window === 'undefined') {
+    process.removeAllListeners('SIGTERM');
```

Comment:
This option would require us to modify the trace/meterprovider to store the cleanup function, and then modify every single test case which uses a trace/meter provider to store and then call the cleanup function. I initially tried this but the task is extremely daunting and would require all future tests to do this as well.

Reply:
Fair enough. I guess this is pretty much what @obecny tried to point at in https://github.com/jonahrosenblum/opentelemetry-js/pull/2/files

I don't have a strong opinion on where this should go, or if it needs to be addressed in this PR. I'm fine with this for now.

Reply:
This approach does not remove the listeners from every other unit test which uses a trace/meterprovider and as a result will still have test cases fail.

Reply:
This is not failing -> https://github.com/open-telemetry/opentelemetry-js/pull/1413
The only failing test was `InMemorySpanExporter.test.ts` due to one initialisation instead of before each test. 
I have also added the missing "cleanup" if for any reason the shutdown will be called directly - you should also remove events that were attached to avoid memory leaking.

Reply:
This makes sense, I have merged your changes. Thank you for your patience I was not fully understanding your implementation until now.